examples/step-30/step-30.cc

```diff
@@ -697,12 +670,13 @@ namespace Step30
                                                   gradient_indicator);
 
     // and scale it to obtain an error indicator.
-    typename DoFHandler<dim>::active_cell_iterator cell =
-                                                     dof_handler.begin_active(),
-                                                   endc = dof_handler.end();
-    for (unsigned int cell_no = 0; cell != endc; ++cell, ++cell_no)
-      gradient_indicator(cell_no) *=
-        std::pow(cell->diameter(), 1 + 1.0 * dim / 2);
+    unsigned int cell_n = 0;
+    for (const auto &cell : triangulation.active_cell_iterators())
+      {
+        gradient_indicator[cell_n] *=
+          std::pow(cell->diameter(), 1 + 1.0 * dim / 2);
+        ++cell_n;
```

Comment:
get rid of `cell_n` altogether. It is equal to `cell->active_cell_index()`, so you don't need the variable and you can all make it a one-line loop again.